File: forcetracker333_ui/src/services/TacticalMapService.js
import http from "../http-common";

class TacticalMapService {
  getAllTacticalMaps(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/tacticalMap/tacticalMaps`, searchDTO);
  }

  get(tacticalMapId) {
    return this.getRequest(`/tacticalMap/${tacticalMapId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/tacticalMap?field=${matchData}`, null);
  }

  addTacticalMap(data) {
    return http.post("/tacticalMap/addTacticalMap", data);
  }

  update(data) {
  	return http.post("/tacticalMap/updateTacticalMap", data);
  }
  
  uploadImage(data,tacticalMapId) {
  	return http.postForm("/tacticalMap/uploadImage/"+tacticalMapId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new TacticalMapService();

File: src/main/java/com/forcetracker333/service/TacticalMapService.java
package com.forcetracker333.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.forcetracker333.domain.TacticalMap;
import com.forcetracker333.dto.TacticalMapDTO;
import com.forcetracker333.dto.TacticalMapSearchDTO;
import com.forcetracker333.dto.TacticalMapPageDTO;
import com.forcetracker333.dto.TacticalMapConvertCriteriaDTO;
import com.forcetracker333.service.GenericService;
import com.forcetracker333.dto.common.RequestDTO;
import com.forcetracker333.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface TacticalMapService extends GenericService<TacticalMap, Integer> {

	List<TacticalMap> findAll();

	ResultDTO addTacticalMap(TacticalMapDTO tacticalMapDTO, RequestDTO requestDTO);

	ResultDTO updateTacticalMap(TacticalMapDTO tacticalMapDTO, RequestDTO requestDTO);

    Page<TacticalMap> getAllTacticalMaps(Pageable pageable);

    Page<TacticalMap> getAllTacticalMaps(Specification<TacticalMap> spec, Pageable pageable);

	ResponseEntity<TacticalMapPageDTO> getTacticalMaps(TacticalMapSearchDTO tacticalMapSearchDTO);
	
	List<TacticalMapDTO> convertTacticalMapsToTacticalMapDTOs(List<TacticalMap> tacticalMaps, TacticalMapConvertCriteriaDTO convertCriteria);

	TacticalMapDTO getTacticalMapDTOById(Integer tacticalMapId);







}





